### PR TITLE
[web] move all build artifacts under web_ui/build

### PR DESCRIPTION
**Pull request by yjbanov**
Put all outputs from the `compile_tests.dart` build step under `web_ui/build` to simplify archiving build artifacts and passing them to test bots. Copy all necessary DEPS dependencies (CanvasKit CIPD package, Skia test images) to `build/` as part of test compilation step. This will allow running tests without running the lengthy `gclient sync -D`.

### CODE REVIEW

File: lib/web_ui/dev/environment.dart [hunk 1]
```diff
@@ -151,7 +151,7 @@ class Environment {
 
   /// Path to the clone of the flutter/goldens repository.
   io.Directory get webUiGoldensRepositoryDirectory => io.Directory(pathlib.join(
-        webUiDartToolDir.path,
+        webUiBuildDir.path,
```
Comment (mdebbar) [inline lib/web_ui/dev/environment.dart:154]: Why this? (I'm curious)

File: lib/web_ui/dev/test_platform.dart [hunk 2]
```diff
@@ -126,15 +120,12 @@ class BrowserPlatform extends PlatformPlugin {
...
         .add(_packageUrlHandler)
 
-        // Serves CanvasKit assets.
-        .add(_canvasKitHandler)
-
         // Serves files from the web_ui/build/ directory at the root (/) URL path.
         //
         // Includes:
         //  * Precompiles .js files for tests
         //  * Sourcemaps
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:127]: Time to update this comment?

File: lib/web_ui/dev/test_platform.dart [hunk 3]
```diff
@@ -126,15 +120,12 @@ class BrowserPlatform extends PlatformPlugin {
...
-        .add(createStaticHandler(env.environment.webUiBuildDir.path))
+        .add(buildDirectoryHandler)
```
Comment (mdebbar) [inline lib/web_ui/dev/test_platform.dart:128]: I like this! Much cleaner than all the custom handlers we had before.

File: lib/web_ui/pubspec.yaml [hunk 4]
```diff
@@ -11,6 +11,7 @@ dependencies:
 
 dev_dependencies:
   archive: 3.1.2
+  file: 6.1.2
```
Comment (mdebbar) [inline lib/web_ui/pubspec.yaml:14]: Do we actually need this? I don't see it used anywhere in this PR.

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/dev/environment.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/dev/environment.dart:154]: To reduce flakiness. It's best if we fetch the repo once during the "build" step, then reuse it in multiple "test" steps. Multiple attempts to fetch the same repo have a higher chance of flaking out due to network issues.

Comment (yjbanov) [reply lib/web_ui/dev/environment.dart:154]: Having said that, this is the overlap that I was referring to. When we go Skia Gold, I'm guessing this won't be necessary.

File: lib/web_ui/dev/test_platform.dart (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/dev/test_platform.dart:127]: Done. I'm not going to list anything. This kind of comments go stale quickly.

File: lib/web_ui/dev/environment.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/dev/environment.dart:154]: Yeah I'm going to clean up all this code once Skia Gold is fully functional!

File: lib/web_ui/pubspec.yaml (hunk 4 above)
Comment (yjbanov) [reply lib/web_ui/pubspec.yaml:14]: Removed. I was tempted to introduce it, but then realized that replacing all the usage of `dart:io` would be scope creep :)

**Pull request merged**